clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5017,8 +5017,9 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
         EmitNounwindRuntimeCall(Fn);
       }
     }
-
-    EmitUnreachable(Loc);
+    // Emit unreachable only for non-sycl devices
+    if (getTarget().getTriple().getSubArch() != getLangOpts().SYCLIsDevice)
```

Comment:
Wow, I guess it should look like this
```suggestion
    if (!getLangOpts().SYCLIsDevice)
```

Reply:
Sorry, I just focused on changing the spir_gen part. Will fix it.